cozy/contexts.py

```diff
@@ -214,7 +214,7 @@ def __init__(self, parent : Context, v : EVar, bag : Exp, bag_pool : Pool):
 
         `v` must not already be described by the parent context.
         """
-        assert v.type == bag.type.elem_type
+        assert v.type == bag.type.elem_type, "{}:{} != {}:{}".format(v, v.type, bag, bag.type.elem_type)
```

Comment:
There is a missing symmetry between the `{}:{}` parts: one prints the type of the expression, and the other prints its element type.

```suggestion
        assert v.type == bag.type.elem_type, "{}:{} != {}:{}".format(v, v.type, bag, bag.type)
```